src/charm.py

```diff
@@ -49,7 +51,20 @@ def __init__(self, *args):
 
         self.framework.observe(self.on.pushgateway_pebble_ready, self._on_pebble_ready)
 
-    def _on_pebble_ready(self, event):
+        self.framework.observe(self.on.push_endpoint_relation_created, self._on_push_relation)
+        self.framework.observe(self.on.push_endpoint_relation_changed, self._on_push_relation)
+
+    def _on_push_relation(self, event: HookEvent) -> None:
+        """Send the push endpoint info."""
+        relation_data = event.relation.data[self.app]
+        relation_data["push-endpoint"] = json.dumps(
+            {
+                "hostname": socket.getfqdn(),
+                "port": self._http_listen_port,
+            }
+        )
```

Comment:
I was wondering if we should also send the path...

AFAIK the default path is `metrics`, but this value can be changed in pushgateway with the [`--web.telemetry-path` option](https://manpages.ubuntu.com/manpages/impish/man1/prometheus-pushgateway.1.html).

Reply:
Good idea! Opened issue #8 for this improvement.